### ISSUE TITLE
Alisim `--write-all` does not write sequences for all internal nodes

### ISSUE THREAD

**Issue by PierreBarrat** [iqtree/iqtree2#372]
Using the following tree
`(((A:0.1)B:0.1)C:0.1)D;`
and the command
`iqtree2 --alisim alignment -m JTT -t tree.nwk --length 10 --out-format fasta --write-all`
produces a file `alignment.fa` with entries for sequences `A` and `D`, but not `B` and `C`. I expected `B` and `C` to be outputted as well since the manual says for `--write-all`: "Enable outputting internal sequences". 

Oddly, if `D` is attributed a branch length (which probably does not make sense) as in
`(((A:0.1)B:0.1)C:0.1)D:0.1;`
then only the sequence for `A` is written. 

*Edit*: I realized that alisim also writes a tree with labels of the internal nodes in `alignment.full.treefile`, and nodes `B` and `C` do not appear in this tree. I imagine that singleton internal nodes are removed from the tree before processing. The behavior observed above is not strictly speaking a bug, but maybe it's worth documenting?  

I take the opportunity to thank you for this great piece of software!

**Comment by bqminh** [comment]
That's correct, it's a feature. Singleton internal nodes are collapsed when loading the tree file. So B and C will disappear. Any internal nodes must have at least two children nodes.

**Comment by trongnhanuit** [comment]
Thanks, @bqminh. Since this is a feature, not a bug. I'll close this issue now.
